commons/src/main/java/com/powsybl/commons/datasource/TarDataSource.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.commons.datasource;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.io.ForwardingOutputStream;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+import org.apache.commons.compress.compressors.xz.XZCompressorInputStream;
+import org.apache.commons.compress.compressors.xz.XZCompressorOutputStream;
+import org.apache.commons.compress.compressors.zstandard.ZstdCompressorInputStream;
+import org.apache.commons.compress.compressors.zstandard.ZstdCompressorOutputStream;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nicolas Rol {@literal <nicolas.rol at rte-france.com>}
+ */
+public class TarDataSource extends AbstractArchiveDataSource {
+
+    public TarDataSource(Path directory, String tarFileName, String baseName, CompressionFormat compressionFormat, String sourceFormat, DataSourceObserver observer) {
+        super(directory, tarFileName, baseName, compressionFormat, ArchiveFormat.TAR, sourceFormat, observer);
+    }
+
+    public TarDataSource(Path directory, String baseName, CompressionFormat compressionFormat, String sourceFormat, DataSourceObserver observer) {
+        super(directory,
+            baseName + sourceFormat + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, compressionFormat, ArchiveFormat.TAR, sourceFormat, observer);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Files are here located in the archive.</p>
+     */
+    @Override
+    public Set<String> listNames(String regex) throws IOException {
+        // Initialize variables
+        Pattern p = Pattern.compile(regex);
+        Set<String> names = new HashSet<>();
+        Path zipFilePath = getArchiveFilePath();
```

Comment:
rename variable "zip" but this is "tar" ?

Reply:
fixed